packages/grafana-runtime/src/components/Chat.tsx

```diff
@@ -0,0 +1,290 @@
+import React, { FunctionComponent, PureComponent, useState } from 'react';
+import { getBackendSrv } from '../services/backendSrv';
+import { TextArea, IconButton, WithContextMenu, MenuItem, MenuGroup } from '@grafana/ui';
+import { getGrafanaLiveSrv } from '../services/live';
+import { isLiveChannelMessageEvent, LiveChannelScope, renderChatMarkdown } from '@grafana/data';
+import { Unsubscribable } from 'rxjs';
+
+export interface ChatProps {
+  contentTypeId: number;
+  objectId: string;
+  actions?: ChatMessageAction[];
+}
+
+export interface User {
+  id: number;
+  name: string;
+  login: string;
+  email: string;
+  avatarUrl: string;
+}
+
+export interface Message {
+  id: number;
+  content: string;
+  created: number;
+  userId: number;
+  user: User;
+}
+
+export interface MessagePacket {
+  event: string;
+  messageCreated: Message;
+}
+
+export interface ChatState {
+  messages: Message[];
+  value: string;
+}
+
+export class Chat extends PureComponent<ChatProps, ChatState> {
```

Comment:
do we need this accessible to plugins (directly) yet?  If not, we should move all frontend changes to:

https://github.com/grafana/grafana/tree/main/public/app/features/chat

and then add @grafana/grafana-edge-squad to CODEOWNERS for that folder.

Reply:
done